proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,83 @@
+
+- This [issue](https://github.com/dotnet/sdk/issues/8254) from 2017 attempting to solve this problsem. It gets several hits a year from customers who are similarly struggling with our toolings inability to handle local deployment.
+- This [internal discussion](https://teams.microsoft.com/l/message/19:ed7a508bf00c4b088a7760359f0d0308@thread.skype/1698341652961?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&parentMessageId=1698341652961&teamName=.NET%20Developer%20Experience&channelName=InfraSwat&createdTime=1698341652961) from a C# team member. They wanted to use VS as the product is shipped to customers and got blocked when we shipped an SDK that didn't have a corresponding MSI and hence VS couldn't load Roslyn anymore.
+- [VS code](https://github.com/dotnet/vscode-csharp/issues/6471) having to adjust to consider local directories for SDK because our resolver can't find them.
+
+## Detailed Design
+The global.json file will support two new properties under the `sdk` object:
+
+- `"additionalLocations"`: this is a list of paths that the host resolver should consider when looking for compatible SDKs. Relative paths will be interpreted relative to the global.json file.
+- `"additionalLocationOnly"`: when true the resolver will _only_ consider paths in `additionalLocations`. It will not consider any machine wide locations (unless they are specified in `additionalLocations`). The default for this property is `false`.
+
+The `additionalLocations` works similar to how multi-level lookup works. It adds additional locations that the host resolver should consider when trying to resolve a compatible .NET SDK. For example:
+
+```json
+{
+    "sdk": {
+        "additionalPaths": [ ".dotnet" ],
+    }
+}
+```
+
+In this configuration the host resolver would find a compatible SDK if it exists in `.dotnet` or a machine wide location.
+
+The values in the `additionalLocations` property can be a relative or absolute path. When a relative path is used it will be resolved relative to the location of global.json. These values also support the following substitutions:
+
+- `"$local$"`: this matches the local installation point of .NET SDK for the current operating system: `%LocalAppData%\Microsoft\dotnet`` on Windows and `$HOME/.dotnet`` on Linux/macOS.
+- `"$machine$"`: this matches the machine installation point of .NET for the current operating system.
+- `%VARIABLE%/$VARIABLE`: environment variables will be substituted. Either the Windows or Unix format can be used here and will be normalized for the operating system the host resolver executes on.
+
+The host resolver will consider the `additionalLocations` in the order they are defined and will stop at the first match. If none of the locations have a matching SDK then it it will fall back to considering machine wide installations (unless `additionalLocationsOnly` is `true`).
```

Comment:
Will it prioritize the PATH first before any of the additional locations? If rollforward is enabled in global.json (or via environment), would it pick the first version that matched even if it wasn't the best match? Example: if the global.json has a 7.0 SDK specified and rollforward is Major and it first finds an 8.0, should it stop or keep looking for a closer match?

Reply:
The order would be `additionalPaths` then how the resolver works today. That is `%PATH%` followed by machine wide installs. I will make that clearer in the doc. 

The intent of the design is that the resolver will stop when it finds the first compatible SDK. For example if `.dotnet` had a compatible SDK but machine wide had a _better_ one then it would pick `.dotnet` because it searched that first. Will make this clearer as well.

Reply:
Ha ... no one caught my "it it" usage. You all are slacking in your reviews!